framework-test/src/main/java/org/checkerframework/framework/test/diagnostics/TestDiagnosticUtils.java

```diff
@@ -201,8 +213,15 @@ public static IPair<String, String> formatJavaxToolString(String original, boole
         String filename = "";
         if (noMsgText) {
             if (!retainAllLines(trimmed)) {
-                int lineSepPos = trimmed.indexOf(System.lineSeparator());
-                if (lineSepPos != -1) {
+
+                // when the source file doesn't use the system default line separator,
+                // System.lineSeparator() can't detect which
+                int lineSepPos = Math.max(trimmed.indexOf("\n"), trimmed.indexOf("\r\n"));
+                if (trimmed.contains("\r\n")) {
+                    trimmed =
+                            trimmed.substring(
+                                    0, lineSepPos + 1); // Include the \r part of the \r\n separator
```

Comment:
I find the numeric logic here hard to follow. Line 219 uses `Math.max`, so isn't `lineSepPos` already on the `\n`? So what does the +1 do?